pkg/machine/compression/gzip.go

```diff
@@ -0,0 +1,37 @@
+package compression
+
+import (
+	"io"
+
+	image "github.com/containers/image/v5/pkg/compression"
+	"github.com/sirupsen/logrus"
+)
+
+type gzipDecompressor struct {
+	genericDecompressor
+	gzReader io.ReadCloser
+}
+
+func newGzipDecompressor(compressedFilePath string) (*gzipDecompressor, error) {
+	d, err := newGenericDecompressor(compressedFilePath)
+	return &gzipDecompressor{*d, nil}, err
+}
+
+func (d *gzipDecompressor) decompress(w WriteSeekCloser, r io.Reader) error {
+	gzReader, err := image.GzipDecompressor(r)
```

Comment:
All other implementations hard-code the decompression implementation instead of indirecting through `c/image/pkg/compression`.

I don’t care which approach is used, but the inconsistency is a bit surprising.